tests/xdc/user_data_replication_test.go

```diff
@@ -218,19 +229,358 @@ func (s *userDataReplicationTestSuite) TestUserDataEntriesAreReplicatedOnDemand(
 		LastProcessedMessageId: -1,
 	})
 	s.NoError(err)
-	numReplicationTasks := len(replicationResponse.GetMessages().ReplicationTasks)
-	s.Equal(numReplicationTasks, numTaskQueues*2+1)
+	var replicationTasks []*replicationspb.ReplicationTask
+	for _, task := range replicationResponse.GetMessages().ReplicationTasks {
+		if task.GetTaskQueueUserDataAttributes().GetNamespaceId() == description.GetNamespaceInfo().Id {
+			replicationTasks = append(replicationTasks, task)
+		}
+	}
+	numReplicationTasks := len(replicationTasks)
+	s.Equal(numReplicationTasks, numTaskQueues*2)
 
-	lastTasks := replicationResponse.GetMessages().ReplicationTasks[numReplicationTasks-numTaskQueues:]
+	lastTasks := replicationTasks[:numTaskQueues]
 	s.Equal(numTaskQueues, len(lastTasks))
 	seenTaskQueues := make(map[string]struct{}, numTaskQueues)
 	// Check the seeded messages
 	for _, task := range lastTasks {
 		s.Equal(enums.REPLICATION_TASK_TYPE_TASK_QUEUE_USER_DATA, task.TaskType)
 		attrs := task.GetTaskQueueUserDataAttributes()
-		s.Equal(description.GetNamespaceInfo().Id, attrs.NamespaceId)
 		seenTaskQueues[attrs.TaskQueueName] = struct{}{}
 	}
 
 	s.Equal(exectedReplicatedTaskQueues, seenTaskQueues)
 }
+
+func (s *userDataReplicationTestSuite) TestUserDataTombstonesAreReplicated() {
+	// Advanced visibility only enabled by default in SQLite
```

Comment:
is that still true? I guess it doesn't matter as long as it runs once in CI

Reply:
Yes, I tested and that's how I ended up adding this condition.
Probably a test configuration issue but I didn't feel like solving it in this PR.